Added support for more themes

eKeiran opened the pull request:
Most of the CSS stylesheets from the original repository are functional, only some that were too different in structure were omitted.

chriskthomas reviewed (changes requested):
Hi,

This is a massive undertaking. I was not expecting so many themes in one go. Thank you! I made some comments on the php files. A few broad notes:

### Move/delete files

* Please put the `Templates` folder in the `src` folder and rename to `templates` (it is case sensitive outside of Windows).
* Delete the js files in every folder except in `PurpleSideOfTheForce`, which actually does something (this can be made to work later)
* Delete all of the index.html files

### PHP code considerations

* Please use globs to make the list of themes and check for the presence of the input in the array rather than the filesystem
* Use `/` for paths instead of `\`

### HTML/CSS considerations (within php files)

* Please avoid inline styles
* Style tags can only exist in the header
* Let's avoid making the html depend on specific themes. In this case (Pride) it is avoidable.

Thank you for the contribution. Overall, I think this is very ambitious and I can see that you have put a lot of work into testing this. Great job! If you would prefer that I wrap this up, just let me know.

chriskthomas commented:
For moving the files btw, you can just do:

```sh
git mv Templates src/templates

# delete index.html files
rm src/templates/*/index.html

# temporarily move index.js from PurpleSideOfTheForce to avoid deletion
git mv src/templates/PurpleSideOfTheForce/index.js foo.js

# delete index.js files
rm src/templates/*/index.js

# Restore preserved js file
git mv foo.js src/templates/PurpleSideOfTheForce/index.js
```

eKeiran commented on src/index.php:
```diff
@@ -93,6 +94,53 @@
+        <div class="mb-3">
+          <label for="theme" class="form-label"><span style="font-weight: 500;">Select Template</span></label>
+          <select id="theme" name="theme" class="form-select">
+            <?php
+            $themes = [
+              "3D Theme",
+              "A mocha, please! (Netlify Form)",
+              "Arcade",
+              "ArghMyEyes",
+              "Art",
+              "BeldiTheme",
+              "CasualBlue",
+              "Cherry-Bloosm",
+              "DarkMode",
+              "DeepSea",
+              "draculaTheme",            
+              "FavouriteTeacherTheme",
+              "GameOn",
+              "GoGreen",
+              "Gridthetic",
+              "Journal",
+              "MagicalMysteryTheme",
+              "Music",
+              "Neon",
+              "Neumorphism",
+              "Nightmare",
+              "NyanCat",
+              "Outrun",
+              "Pastel Dreams",
+              "Pink",
+              "Pride",
+              "PurpleSideOfTheForce",
+              "RedZone",
+              "Retro Game",
+              "Shades Of Purple",
+              "Sixties",
+              "Sky",
+              "Standard"
+            ];
```
Alright! I'll replace it with glob

eKeiran commented on src/api.php:
```diff
@@ -23,37 +23,68 @@
 
     <?php if (!empty($_POST["description"])) {?>
-    <p><?=$_POST["description"]?></p>
+        <div class="container"><p style="text-align: center; font-size: 1.25rem;"><?=$_POST["description"]?></p>  </div>
```
Ooh yes, cool!

eKeiran commented on src/api.php:
```diff
@@ -23,37 +23,68 @@
+    <?php
+if (isset($_POST['theme'])) {
+    $selectedTheme = $_POST['theme'];
+    $themeFolderPath = '../Templates/' . $selectedTheme;
```
I see! Thanks for letting me know. Yep, I'll do it using glob soon

eKeiran commented:
Completed the requested changes:

1. Moving/Deleting the requested files.
2. PHP: Using Globs to make an array of themes and validating input against the array for dealing with the security risk.
3. CSS/HTML: Removed inline styles and implemented body::before for the Pride theme.

Additional changes:
PHP:
1. Added favicon support.
2. Implemented audio functionality for the PurpleSideOfTheForce theme.

chriskthomas commented:
Thanks! I don't think I can merge this directly. I think I'll just roll these changes into the repo manually and set you as the author of the commit

chriskthomas commented:
Hi @eKeiran. I think it might be cleaner to put all the styles in a separate repo and keep the php here. I put your extensive css edits over here chriskthomas/linkfree-themes#2 in a separate repo.

I will then merge the php edits over here and make the css files available through a git submodule.

chriskthomas commented:
Merged in the form of #13
